### ISSUE TITLE
Design Discussion

### ISSUE THREAD

**Issue by edmundjarvis** [SocialistRA/socialistra.github.io#1]
Let's talk layout, logo's, etc.

**Comment by JasonLeeH** [comment]
I can certainly make logos. I've made some patches already. First, let's decide what kind of page content will be displayed and then work backwards from there.

From the Reddit post:

> Page 1: Basic mission statement/news ticker on the first page
> Page 2: Drop down list of all chapters/their facebook pages/subreddits/contact information.
> Page 3: Some kind of typeform or google form we can use to form an organizing committee or something of that nature.

So I'm thinking Page 1 can go on the homepage and then at the bottom there can be a "Find Your Local Chapter" link which takes you to a page where Page 2 is shown. And on _that_ page, there will be a DIV which says something like "Can't Find Your City? Start Your Own Chapter!", which will link to Page 3.

**Comment by edmundjarvis** [comment]
I like your patches, especially the workers front sight post and the LGBTQ pride. The workers front site might make a good favicon if scaled down.

I agree with the 3 pages there for a start. Getting the local chapters up is probably highest priority.

I'm planning on writing up some explanatory pages such as 'What is Socialism' and 'Why the SRA?' - going into the basics of what socialism is (since there's so much confusion in the US) and why someone would want to join the SRA instead of the NRA.

**Comment by JasonLeeH** [comment]
Use this font for the title: https://fonts.google.com/specimen/Rokkitt

![Title](https://i.imgur.com/5zoicjv.png)

Also, it might look better if we just use a black star for now:
`<img width="30px" src="https://i.imgur.com/QdsGQVc.png" style="padding-bottom:10px">`

![Revised Title](https://i.imgur.com/LalRdEf.png)

**Comment by edmundjarvis** [comment]
Good pick. Any opinion on font weight? I tend to like lighter weight, but thought the medium or heavier weights might look good here.

**Comment by JasonLeeH** [comment]
I think it's fine the way it is for now. I don't know if you saw, but you should add a bottom padding to the logo image of 10px in order to horizontally align the logo and the title.

**Comment by edmundjarvis** [comment]
Fixed.

**Comment by JasonLeeH** [comment]
I'm taking it upon myself to write some OG site template code. My patches on the subreddit are receiving the most widespread support so I've decide to go ahead and create a logo. I literally *just* got this tiny little bit of design work done so I'll push the code in the morning. You'll need to add me as a contributor to the repository so I can push instantly. Until then, check this out:

![Site Preview](https://jason.sx/dev/demo.png)

**Comment by JasonLeeH** [comment]
Here is the logo on a couple of patches. I think it looks tight.

![Guard Patch](https://i.imgur.com/abzrQYV.jpg)

![Field Patch](https://i.imgur.com/wV1BBjH.jpg)

 - **Meaning** - Three rounds of equal length, representing comrades in solidarity, are centered. On the outside are sheaves of wheat growing from a rotary gear, representing how industry is vital to abundance. All of this is presented under a socialist star, representing how soldiers will fight under the socialist banner. Further, there is a symbiotic relationship between the soldiers and the industry: The industry will feed and supply the soldier and the soldier will defend and protect the people's industry.

**Comment by edmundjarvis** [comment]
That's awesome. Some of the other logos and patches floating around were very Stalinist, I think this one is perfect for representing the whole left.

I'll get you added to the SRA organization here, so you should be able to push.

**Comment by edmundjarvis** [comment]
Design is looking great so far. I think we should still use a static site generator such as Jekyll though, as it'll be very helpful as the site gets larger with more pages.

I can go ahead and start converting your work into reusable templates if that's ok and doesn't throw you off.

**Comment by JasonLeeH** [comment]
I'm going to use PHP to turn the template into a CMS system with a database so new pages can be created and displayed on the fly.

**Comment by JasonLeeH** [comment]
If you want to create page templates using the code I've made so far, go ahead. But keep them in their own distinct directory outside of the root directory.

**Comment by JasonLeeH** [comment]
I'm going to use PHP to turn the template into a CMS system with a database so new pages can be created and displayed on the fly. If you want to create page templates using the code I've made so far, go ahead. But keep all of the new code (especially CSS) in their own distinct directory outside of the root directory.

Also, if you want to upload the new code to the domain, you'll notice a lot of new changes.

**Comment by edmundjarvis** [comment]
Sorry must have been some confusion. Right now this is hosted straight out of github, but that only works with static sites. For PHP to work we'll have to move this all to a dedicated server. We can use yours or I've got a few Amazon cloud instances we can use.

**Comment by JasonLeeH** [comment]
Email me and I'll give you my dedicated server IP so you can set up the A and CNAME records on the domain.

_snip_

After the IP switch, you'll need to reconfigure Cloudflare (probably by removing the site and starting over the setup process).

**Comment by JasonLeeH** [comment]
Looks like the site is being hosted on Github again? If you could change the records back, that'd be great. I need PHP to work.

EDIT: Or, if you would like, just decrease the distance between cache flushes. It's nice to see code edits update in-browser.

**Comment by JasonLeeH** [comment]
The website isn't resolving to the correct IP address.

**Comment by edmundjarvis** [comment]
socialistra.org is working fine for me. www.socialistra.org wasn't resolving right, checking settings now.

**Comment by JasonLeeH** [comment]
The former auto-redirects to the latter, which fails due to HSTS.

**Comment by edmundjarvis** [comment]
Yeah I've got the address for www. updated, might take a while to clear old setting.

**Comment by capito-tar-gz** [comment]
I know I'm coming into this late, but if you make the site a static site with node.js using something like metalsmith, adding pages could be as easy as making a new markdown file. That could be more accessible than Jekyll for developers who don't want to deal with Ruby and more secure than a PHP CMS.

Sorry if I'm being a bother! I was trying to click links on the live site but could only navigate by seeing the folder structure in this repository.
